luarules/gadgets/game_team_power_watcher.lua

```diff
@@ -0,0 +1,514 @@
+    {techLevel = 2, threshold = 90000},
+    {techLevel = 2.5, threshold = 230000},
+    {techLevel = 3, threshold = 350000},
+    {techLevel = 3.5, threshold = 475000},
+    {techLevel = 4, threshold = 600000},
+    {techLevel = 4.5, threshold = 725000}
+}
+
+for _, teamID in ipairs(teamList) do
+    Spring.Echo("teamID", teamID)
+    local teamLuaAI = Spring.GetTeamLuaAI(teamID)
+    if (teamLuaAI and string.find(teamLuaAI, "ScavengersAI")) then
+        scavengerTeam = teamID
+    elseif (teamLuaAI and string.find(teamLuaAI, "RaptorsAI")) then
+        raptorTeam = teamID
+    elseif select (4, Spring.GetTeamInfo(teamID, false)) then
+        aiTeams[teamID] = true
+    elseif teamID == tonumber(teamList[#teamList]) then
+        neutralTeam = teamID
+    else
+        humanTeams[teamID] = true
+    end
+end
+
+--assign team powers/peak powers to 0 to prevent nil
+for _, teamNumber in ipairs(teamList) do
+    teamPowers[teamNumber] = 0
+    peakTeamPowers[teamNumber] = 0
+end
+
+
+
+function gadget:UnitFinished(unitID, unitDefID, unitTeam)
+    unitsWithPower[unitID] = {unitID = unitID, power = UnitDefs[unitDefID].power, team = unitTeam}
+    teamPowers[unitTeam] = (teamPowers[unitTeam] + UnitDefs[unitDefID].power) or UnitDefs[unitDefID].power
+    if teamPowers[unitTeam] and peakTeamPowers[unitTeam] < teamPowers[unitTeam] then
+        peakTeamPowers[unitTeam] = teamPowers[unitTeam]
+    end
+end
+
+function gadget:UnitDestroyed(unitID, unitDefID, unitTeam)
+    if unitsWithPower[unitID] then
+        unitsWithPower[unitID] = nil
+    end
+    if UnitDefs[unitDefID].power then
+        if teamPowers[unitTeam] then
+            if teamPowers[unitTeam] <= UnitDefs[unitDefID].power then
+                teamPowers[unitTeam] = 0
+            else
+                teamPowers[unitTeam] = teamPowers[unitTeam] - UnitDefs[unitDefID].power
+            end
+        end
+    end
+end
+
+--handles capture events on units already added to unitsWithPower by UnitFinished
+function gadget:MetaUnitAdded(unitID, unitDefID, unitTeam)
+    if unitsWithPower[unitID] then
+        local oldTeam = unitsWithPower[unitID].team
+
+        unitsWithPower[unitID] = {unitID = unitID, power = UnitDefs[unitDefID].power, team = unitTeam}
+        teamPowers[unitTeam] = (teamPowers[unitTeam] + UnitDefs[unitDefID].power) or UnitDefs[unitDefID].power
+
+        if teamPowers[oldTeam] <= UnitDefs[unitDefID].power then
+            teamPowers[oldTeam] = 0
+        else
+            teamPowers[oldTeam] = teamPowers[unitTeam] - UnitDefs[unitDefID].power
+        end
+
+        if teamPowers[unitTeam] and peakTeamPowers[unitTeam] < teamPowers[unitTeam] then
+            peakTeamPowers[unitTeam] = teamPowers[unitTeam]
+        end
+    end
+end
+
+--Returns the power of the input teamID as a number.
+local function teamPower(teamID)
+    for id, power in pairs(teamPowers) do
+        if id == teamID then
+            return power
+        end
+    end
+    return 0
```

Comment:
```suggestion
   return teamPowers[teamID]
```
If a value in a table can be looked up by ID, there is no need to iterate. If called with a team ID that doesn't exist, semantically it should return nil (this is an error condition) rather than 0 (this is a default condition).